app/src/main/java/ch/epfl/sdp/map/fragment/MapFragment.java

```diff
@@ -435,6 +438,38 @@ private void callOnMapVisible() {
         }
     }
 
+    private void askGPSPermission(){
```

Comment:
A possibility would be to extract a method for the AlertDialog.Builder. Though in my opinion you don't necessarily have to, since this method is clear enough
It's up to you!